float8_experimental/float8_tensor.py

```diff
@@ -123,6 +132,9 @@ def __tensor_flatten__(self):
 
     @staticmethod
     def __tensor_unflatten__(inner_tensors: Dict, metadata):
+        # TODO(TBD): this seems unused, and it's out of date after
```

Comment:
I was looking into this today and I do think that this is used it is just that in full graph compile land we get away with this not being a problem we should update this to the new signature

Reply:
hmm, do you have more context?

If I understand the current state correctly:
1. the currently checked in code is broken because the signature of __tensor_unflatten__ is no longer valid after https://github.com/pytorch/pytorch/pull/114311
2. if this code path is exercised by dynamo, it will crash because of (1)
3. all tests pass, so I'm guessing this is dead code at the moment

Reply:
@bdhirsh I imagine that although this is broken we are not encountering this because we can full graph compile and the tensor subclass vanishes from aot, is that correct or am I missing something? So although this is not exercised today if we decided to expose float8tensor subclass as more of a first ux and they entered compiled functions we would need this again

Reply:
Yep - I think that if your subclass is just a temporary that poofs during tracing (so not a graph input or output), then we won't end up hitting the flatten/unflatten calls at all. We'll definitely need it though if/when the subclass ever becomes a graph input/output.

This can happen if you make the subclass a module parameter directly, **but** it could also just happen via graph breaks, if there happens to be a graph break after we create the float8 tensor but before it vanishes (which to be fair would probably be bad for the level of perf that float8 wants anyway).

Reply:
thanks @bdhirsh  for explaining.  I'll remove the TODO from this PR, and looks like we should fix the signature and add a unit test with the conditions you described.

Reply:
put up https://github.com/pytorch-labs/float8_experimental/pull/166, it doesn't work yet, @bdhirsh  any thoughts on the error message in that PR?

Reply:
yep its in the test plan (not in CI)!

Reply:
missed it the first time https://gist.github.com/vkuzo/ba98a01a459fb9c966f167d8ecca1780#file-gistfile0-txt-L137.

Hmm, this looks like the same error @drisspg and I noticed a month or two ago, where (somehow) we end up with multiple fake modes floating around. This probably happens somewhere in dynamo, since dynamo is responsible for creating the FakeTensorMode.

This bug might be pretty involved, so just checking - "get float8tensor working with graph breaks" a blocker? (Regardless I agree that we should make it work)

Reply:
not technically a blocker, but would be really nice for it to work